pyloopkit/dose_entry.py

```diff
@@ -23,28 +23,32 @@ def net_basal_units(type_, value, start, end, scheduled_basal_rate):
     end -- datetime object representing end of dose
     scheduled_basal_rate -- the rate scheduled during the time the dose was
                             given (0 for boluses)
+    delivered_units -- units actually delivered by pump
 
     Output:
     Bolus amount (if a bolus), or basal units given, net of whatever the
-    schedule basal is
+    scheduled basal is
     """
     MINIMUM_MINIMED_INCREMENT = 20
 
     if type_ == DoseType.bolus:
-        return value
+        return delivered_units if delivered_units is not None else value
```

Comment:
I'm wondering if we can just make it a requirement that all doses have delivered units. Might simplify the code. Loop needed to support it optionally as PumpManagers weren't all updated at the same time, but even Loop code could be cleaned up now, I think.

Reply:
I think that is fine. @novalegra Is that a quick adjustment?

Reply:
Sure, I'll do that now

Reply:
Actually, @scaubrey, I'm realizing that doing so would break almost EVERY test for PyLoopKit. Is this a real need at the moment? I would have to re-do a significant portion of the tests, and those tests would not be the same as those in Loop.

Reply:
@novalegra Nope, it is not a real need right now. I think as long as @ps2 feels it's ok to leave as it is and address later, we can wait.